docs/Hooks.md

```diff
@@ -141,6 +141,48 @@ fastify.addHook('onSend', (request, reply, payload, next) => {
 
 Note: If you change the payload, you may only change it to a `string`, a `Buffer`, a `stream`, or `null`.
 
+### Respond to a request from an hook
+If need you can respond to a request before you reach the route handler, an example could be an authentication hook. If you are using `onRequest` or a middleware just use `res.end`, if you are using the `preHandler` hook use `reply.send`. Remember to always call `next` if you are using the standard hook api, if you are working with *async* hooks it will be done automatically by Fastify.
+```js
+// standard api
+fastify.addHook('preHandler', (request, reply, next) => {
+  reply.send({ hello: 'world' })
+  next()
+})
+
+// async api
+fastify.addHook('preHandler', async (request, reply) => {
+  reply.send({ hello: 'world' })
+})
+```
+
+If you want to respond with a stream, you should make sure that you don't call `next` before the response has finished.
+
+```js
+// standard api
+fastify.addHook('onRequest', (req, res, next) => {
+  const stream = fs.createReadStream('some-file', 'utf8')
+  stream.pipe(res)
+  res.once('finish', next)
+})
```

Comment:
Be careful that this code is substantially wrong because it will leak a file descriptor if `stream` errors. How about using `pump`?